python-frontend/src/main/java/org/sonar/python/semantic/FunctionSymbolImpl.java

```diff
@@ -225,6 +227,15 @@ public InferredType declaredReturnType() {
     return declaredReturnType;
   }
 
+  @Override
+  public Optional<ClassSymbol> returnTypeHint() {
+    if (declaredReturnType instanceof RuntimeType) {
```

Comment:
I think we shouldn't return the typeClass of RuntimeType. I think we should instead return the symbol of the "TypeAnnotation" tree